design-toolkit/components/src/components/drawer/index.tsx

```diff
@@ -227,8 +247,42 @@ function DrawerHeaderTitle({ className, level, ...rest }: DrawerTitleProps) {
 }
 DrawerHeaderTitle.displayName = 'Drawer.Title';
 
-function DrawerHeader({ className, ...rest }: ComponentPropsWithRef<'header'>) {
-  return <Header {...rest} className={header({ className })} />;
+function DrawerHeader({
+  className,
+  title,
+  children,
+  ...rest
+}: ComponentPropsWithRef<'header'>) {
+  const { stack } = useContext(ViewStackContext);
+  const level = stack.length > 1 ? 6 : 1;
```

Comment:
NIT: There's no perfect semantic solution here since headings could be anything in regards to the overall page hierarchy, but switching between 1 and 6 feels wrong